src/Commands.vala

```diff
@@ -1226,7 +1234,11 @@ public class SetRawDeveloperCommand : MultipleDataSourceCommand {
     }
 
     public override void undo_on_source (DataSource source) {
-        Photo? photo = source as Photo;
+        Photo? photo = null;
+        if (source is Photo) {
+            photo = source as Photo;
+        }
```

Comment:
Isn't this what `as` already does as compared to an unsafe cast? It'll return `null` if the type doesn't match?

Reply:
Yes, but it is also giving those terminal criticals.  I can rewrite using unsafe casts and using `source is Photo` instead of `photo != null` if you like.